packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/Drawing/DrawingContainer.js

```diff
@@ -1,143 +1,42 @@
-  // TEMP CODE UNTIL COORDINATE STREAM
-  tempMarkCreation () {
-    const { activeDrawingTool, activeStepTasks } = this.props
-
-    if (activeStepTasks && activeStepTasks.length > 0) {
-      const [activeDrawingTask] = activeStepTasks.filter(task => task.type === 'drawing')
-      const tool = activeDrawingTask.tools[activeDrawingTool]
-      if (tool.type === 'line') {
-        const coordinates = { x1: (Math.floor(Math.random() * 400)), y1: (Math.floor(Math.random() * 400)), x2: (Math.floor(Math.random() * 400)), y2: (Math.floor(Math.random() * 400)) }
-        this.finishMark(coordinates)
-      }
-      if (tool.type === 'point') {
-        const coordinates = { x: (Math.floor(Math.random() * 400)), y: (Math.floor(Math.random() * 400)) }
-        this.finishMark(coordinates)
-      }
-    }
-  }
-
-  setActiveMark () {
-    const { activeDrawingTool, activeStepTasks } = this.props
-
-    if (activeStepTasks && activeStepTasks.length > 0) {
-      const [activeDrawingTask] = activeStepTasks.filter(task => task.type === 'drawing')
-      this.setState({
-        activeMark: {
-          id: cuid(),
-          tool: activeDrawingTask.tools[activeDrawingTool]
-        }
-      })
-    }
-  }
-
-  finishMark (coordinates) {
-    if (!coordinates) return null
-
-    const { activeDrawingTool, activeStepTasks } = this.props
-    const { activeMark, marks } = this.state
-
-    const [activeDrawingTask] = activeStepTasks.filter(task => task.type === 'drawing')
-
-    const newMark = {
-      id: cuid(),
-      tool: activeDrawingTask.tools[activeDrawingTool]
-    }
-
-    const finishedMark = Object.assign(activeMark, { coordinates })
-
-    const newMarks = new Map(marks)
-    newMarks.set(finishedMark.id, finishedMark)
-
-    this.setState({ activeMark: newMark, marks: newMarks })
-  }
-
   render () {
-    const { activeMark, marks } = this.state
-    if (!activeMark) return null
-
-    const MarkComponent = getDrawingTool(activeMark.tool.type)
-    // TODO: add error handling if don't get requested tool type
-
+    const { activeDrawingTask, activeMark, marks } = this.props
     return (
       <>
-        <MarkComponent
-          key={activeMark.id}
-          active
-          finishDrawing={this.finishMark}
-          tool={activeMark.tool}
-        />
-        {marks.size > 0 && Array.from(marks, ([id, marking]) => {
-          if (!marking) {
-            return null
-          }
+        {Array.from(marks, ([id, mark]) => {
+          if (!mark.coordinates) return null
+
+          const tool = activeDrawingTask.tools[mark.toolIndex]
+          const MarkingComponent = getDrawingTool(tool.type)
 
-          const MarkingComponent = getDrawingTool(marking.tool.type)
           return (
             <MarkingComponent
               key={id}
-              coordinates={marking.coordinates}
-              tool={marking.tool}
+              active={mark === activeMark}
+              coordinates={mark.coordinates}
+              tool={tool}
```

Comment:
You can probably store the tool on the Mark model, then reference `mark.tool` here.